source/develop/release-management/features/external-network-provider.html.md

```diff
@@ -0,0 +1,525 @@
+
+*   Email: <mmirecki@redhat.com>
+
+### Benefit to oVirt
+
+oVirt VMs will be able to use networks definded by external providers.
+oVirt will provide an API for the external providers (vendors of Software Defined
+Networking products) allowing them to easily integrate with oVirt.
+
+
+## Detailed Description
+
+Many organizations use centralized network management systems to handle all
+their networking, and would like them to manage the network topology of
+their oVirt environments. 
+Currently the only option of using external networks in oVirt is to use the
+OpenStack Neutron integration, which allows the import of OpenStack Neutron
+networks and the provisioning of VM's connected to these networks. 
+This feature is however very specific to OpenStack Neutron and tied to its Linux bridge and OVS plugins.. 
+We would like to extend this functionality to other external network providers,
+by making it simpler, more general and less dependant on OpenStack Neutron features. 
+The result should be an API over which oVirt can communicate with external
+network management systems, and use the networks defined in them in provisioned VMs.
+
+For information about OpenStack Neutron integration please refer to:
+[OSN Integration](http://www.ovirt.org/develop/release-management/features/network/osn-integration/)
+[Detailed OSN Integration](http://www.ovirt.org/develop/release-management/features/network/detailed-osn-integration/)
+
+### Differences from OpenStack Neutron Integration
+
+*   hiding of UI features related to OpenStack Neutron, such as tenants and OpenStack Neutron driver details
+![](external_network_provider_import_dialog_changes.png "fig:external_network_provider_import_dialog_changes.png")
+
+*   simplified REST API. Base on the OpenStack Neutron API, but simplified to contain
+only a subset of its schema which is relevant to the external network providers.
+This will require a rework of the UI and REST API layer.
+
+*   read-only mode for external network
+
+*   (optionally) read-only mode for subnets
+
+*   VIF driver - the implementation of the VIF driver is now completely the responsibility 
+of the external network provider
+
+*   No special handling of OpenStack Neutron agent. - since the VIF driver implementation is provider dependent,
+the OpenStack Neutron agent is not a part of the solution anymore (unless the provider
+decides otherwise)
+
+*   VIF driver reference implementation - a sample VIF driver showing a simple implementation is provided
+
+
+### Main functional parts of the system
+The API for external network providers consists of 3 main parts:
+
+1. External network provider administration. 
+  This allows to perform admistrative tasks:
+  * connecting to a new network provider (External Providers -> Add)
+  * importing networks from an external provider (External Providers -> Networks Subtab -> Import)
+  * removing a network (Networks -> Remove)
+  * adding a new subnet pool (Networks -> Subnets tab -> New)
```

Comment:
the last two tasks are allowed only on read-write providers. such providers also allow adding a new network on them.